src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -74,8 +76,16 @@ The .NET Foundation licenses this file to you under the MIT license.
       <NativeFramework Include="GSS" />
     </ItemGroup>
 
+    <Message Text="IlcPath = $(IlcPath)" />
+    <Message Text="IlcHostPackagePath = $(IlcHostPackagePath)" />
+
+    <Exec Command="$(IlcPath)/source_code/libs/System.Globalization.Native/build-nativeaot-shim-standalone.sh $(IlcPath)/../ $(IntermediateOutputPath)" Condition="'$(LocalSystemGlobalizationNative)' != '' and '$(IlcPath)' != ''"/>
+
+    <Exec Command="$(IlcHostPackagePath)/source_code/libs/System.Globalization.Native/build-nativeaot-shim-standalone.sh $(IlcHostPackagePath)/ $(IntermediateOutputPath)" Condition="'$(LocalSystemGlobalizationNative)' != '' and '$(IlcHostPackagePath)' != '' and '$(IlcPath)' == ''"/>
+
     <ItemGroup>
       <LinkerArg Include="@(NativeLibrary)" />
+      <LinkerArg Include="@(StaticICULibs)" Condition="'$(LocalSystemGlobalizationNative)' != ''" />
```

Comment:
I would move this below where we have the other libraries (there is line for `licucore` on Mac).